queries/ja.py

```diff
@@ -0,0 +1,297 @@
+
+    return res
+
+
+_MOBILE_FIRST_NUM = "678"
+
+
+def _best_number(item: Dict) -> Optional[str]:
+    """ Return best phone number, given a result item from ja.is API """
+    phone_num = item.get("phone")
+    add_nums = item.get("additional_phones")
+    if not phone_num and not add_nums:
+        return None
+
+    # First, see if the canonical phone number is a mobile phone number
+    # Those should be preferred
+    if phone_num and "number" in phone_num:
+        if phone_num.get("mobile") == True:
+            return phone_num["number"]
+
+    # Otherwise, try the first mobile number we find in add. phone numbers
+    if add_nums:
+        for pn in add_nums:
+            if pn and "number" in pn and pn.get("mobile") == True:
+                return pn["number"]
+
+    # OK, didn't find any mobile numbers. Just return canoncial number
+    return phone_num.get("number") if phone_num else None
+
+
+def _answer_phonenum4name_query(q: Query, result):
+    """ Answer query of the form "hvað er síminn hjá [íslenskt mannsnafn]?" """
+    res = query_ja_api(result.qkey)
+
+    nþgf = NounPhrase(result.qkey).dative or result.qkey
+
+    # Verify that we have a sane response with at least 1 result
+    if not res or not res.get("people") or not res["people"].get("items"):
+        return gen_answer("Ekki tókst að fletta upp {0}.".format(nþgf))
+
+    # Check if we have a single canonical match from API
+    single = len(res["people"]["items"]) == 1
+    allp = res["people"]["items"]
+    first = allp[0]
+    fname = first["name"]
+    if not single:
+        # Many found with that name, generate smart message asking for disambiguation
+        one_name_only = len(result.qkey.split()) == 1
+        msg = "Það fundust margir með það nafn. Prufaðu að tilgreina {0}heimilisfang".format(
+            "fullt nafn og " if one_name_only else ""
+        )
+        # Try to generate example, e.g. "Jón Jónssón á Smáragötu"
+        for i in allp:
+            try:
+                street_nf = i["address_nominative"].split()[0]
+                street_þgf = i["address"].split()[0]
+                msg = msg + " t.d. {0} {1} {2}".format(
+                    fname, iceprep_for_street(street_nf), street_þgf
+                )
+                break
+            except Exception as e:
+                print("Exception " + str(e))
+                continue
+        return gen_answer(msg)
+
+    # Scan API call result, try to find the best phone nuber to provide
+    phone_number = _best_number(first)
+    if not phone_number:
+        a = "Ekki tókst að fletta upp símanúmeri hjá {0}".format(nþgf)
+        return gen_answer(a)
+
+    # Sanitize number and generate answer
+    phone_number = phone_number.replace("-", "").replace(" ", "")
+    answ = phone_number
+    fn = NounPhrase(fname).dative or fname
+    voice = "Síminn hjá {0} er {1}".format(fn, " ".join(list(phone_number)))
+
+    q.set_context(dict(phone_number=phone_number, name=fname))
```

Comment:
We should store the name->phone number map in the client data, and use the map in disambiguation.